### PR TITLE
Use join instead of "+"

### LINKED ISSUE #453: Extra "/" preventing exploration of more than one "subdirectory"
I'm getting a double slash introduced when I navigate using the intake gui to explore an s3 bucket, which then does not show the next level:

If I manually change `//` to `/` it works.

### PR DESCRIPTION
**Pull request by rsignell-usgs**
Fixes problem drilling more than one directory deep

Fixes #453

### CODE REVIEW

File: intake/gui/catalog/add.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def move_down(self, *events):
                 fn = event.new[0]
                 if fn.endswith('/'):
                     if self.path_text.value:
-                        self.path_text.value = self.path_text.value + '/' + fn
+                        self.path_text.value = os.path.join(self.path_text.value, fn)
```
Comment (martindurant) [inline intake/gui/catalog/add.py]: This will fail on Windows, where the separator is "\\". Perhaps `posixpath.join`?

Comment (rsignell-usgs) [reply intake/gui/catalog/add.py]: @martindurant, isn't this the situation?  This is what I get on Windows:
```python
a = 's3://bucket/key1/'
b = 'key2/'
print(os.path.join(a,b))

's3://bucket/key1/key2/'
```

Comment (martindurant) [reply intake/gui/catalog/add.py]: What if `a = 'bucket'` ?

Comment (rsignell-usgs) [reply intake/gui/catalog/add.py]: Can that be the case?  I thought the GUI puts a "/" on the end.

Comment (martindurant) [reply intake/gui/catalog/add.py]: I suppose you are right, if the path ends with "/", it amounts to the same thing. This is a surprise to me! I never work with windows paths for a reason.

**Pull request merged**